File: api/resource/src/main/java/com/mengyunzhi/measurement/Service/AccuracyService.java
package com.mengyunzhi.measurement.Service;

import com.mengyunzhi.measurement.entity.Accuracy;

/**
 * Created by panjie on 17/6/28.
 * 精度
 */
public interface AccuracyService {
    // 保存
    Accuracy save(Accuracy accuracy);

    Accuracy getOneSavedAccuracy();
    static Accuracy getOneAccuracy() {
        Accuracy accuracy = new Accuracy();
        return accuracy;
    }
    void deleteById(Long id);

    /**
     * 获取某个器具类别下的所有精确度
     * @param instrumentTypeId 器具类别id
     * @return
     * panjie
     */
    Iterable<Accuracy> getAllByInstrumentTypeId(Long instrumentTypeId);

}

File: Webapp/app/scripts/services/accuracyservice.js
'use strict';

/**
 * @ngdoc service
 * @name webappApp.AccuracyService
 * @description 准确度等级Service层
 * # AccuracyService
 * Service in the webappApp.
 */
angular.module('webappApp')
    .service('AccuracyService', ['$http', 'CommonService', function ($http, CommonService) {
        var self = this;

        /**
         * 获取某个器具类别下的所有实体
         * @param instrumentTypeId 器具类别id
         * @param callback
         * panjie
         */
        self.getAllByInstrumentTypeId = function(instrumentTypeId, callback) {
            var url = '/Accuracy/getAllByInstrumentTypeId/' + instrumentTypeId;
            $http.get(url)
                .then(function success(response){
                    if (callback) {callback(response.data);}
                }, function error(response){
                    CommonService.httpError(response);
                });
        };

        // 添加
        self.save = function (data, callback) {
            $http.post('/Accuracy/', data)
                .then(function success(response) {
                    if (response.status === 201) {
                        callback(response.data);
                    }
                }, function error(response) {
                    CommonService.httpError(response);
                });
        };

        // 删除
        self.deleteById = function (id, callback) {
            var url = '/Accuracy/' + id;
            $http.delete(url)
                .then(function success(response) {
                    if (callback) {
                        callback(response);
                    }
                }, function error(response) {
                    CommonService.httpError(response);
                });
        };

        return {
            getAllByInstrumentTypeId: self.getAllByInstrumentTypeId,
            save: self.save,
            deleteById: self.deleteById
        };
    }]);
